src/Chip/Chip.js

```diff
@@ -0,0 +1,338 @@
+    }
+  };
+
+  handleMouseEnter = (event) => {
+    this.handleFocus();
+    this.props.onMouseEnter(event);
+  };
+
+  handleMouseDown = (event) => {
+    //only listen to left clicks
+    if (event.button === 0) {
+      event.stopPropagation();
+      this.setState({clicked: true});
+      this.props.onMouseDown(event);
+    }
+  };
+
+  handleMouseLeave = (event) => {
+    if (!this.refs.button.isKeyboardFocused()) {
+      this.handleBlur();
+    }
+    this.props.onMouseLeave(event);
+  };
+
+  handleMouseUp = (event) => {
+    this.setState({clicked: false});
+    this.props.onMouseUp(event);
+  };
+
+  handleTouchStart = (event) => {
+    event.stopPropagation();
+    this.setState({clicked: true});
+    this.props.onTouchStart(event);
+  };
+
+  handleTouchEnd = (event) => {
+    this.setState({clicked: false});
+    this.props.onTouchEnd(event);
+  };
+
+  render() {
+    const {prepareStyles} = this.context.muiTheme;
+    const styles = getStyles(this.props, this.context, this.state);
+
+    const chipStyle = prepareStyles(Object.assign(styles.chip, this.props.chipStyle));
+    const rootStyle = Object.assign(styles.root, this.props.style);
+    const labelStyle = prepareStyles(Object.assign(styles.label, this.props.labelStyle));
+
+    const {className, deletable, ...other} = this.props;
+    let {avatar, children, label} = this.props;
+
+    let deleteIcon = deletable ?
+      <DeleteIcon
+        color={styles.closeIcon.color}
+        style={styles.closeIcon}
+        onTouchTap={this.handleCloseIconTouchTap}
+      /> :
+      null;
+
+    // If the`avatar` prop isn't set, check if the first child is an Avatar
+    if (!avatar) {
+      const childCount = React.Children.count(children);
+
+      if (childCount > 1) {
+        children = React.Children.toArray(children);
+
+        if (React.isValidElement(children[0]) && children[0].type === Avatar) {
+          avatar = children[0];
+          label = children[1];
+        }
+      }
+    }
+
+    if (React.isValidElement(avatar)) {
+      avatar = React.cloneElement(avatar, {
+        style: Object.assign(styles.avatar, avatar.props.style),
+        size: 32,
+      });
+    } else if (avatar !== null) {
+      avatar = <Avatar src={avatar} style={styles.avatar} size={32} />;
+    }
+
+    label = label || children;
+
+    return (
+      <EnhancedButton
+        ref="button"
+        className={className}
+        disableTouchRipple={true}
+        disableFocusRipple={true}
+        onBlur={this.handleBlur}
+        onFocus={this.handleFocus}
+        onKeyDown={this.handleKeyDown}
+        onMouseDown={this.handleMouseDown}
+        onMouseEnter={this.handleMouseEnter}
+        onMouseLeave={this.handleMouseLeave}
+        onMouseUp={this.handleMouseUp}
+        onTouchEnd={this.handleTouchEnd}
+        onTouchStart={this.handleTouchStart}
+        onKeyboardFocus={this.handleKeyboardFocus}
+        style={rootStyle}
+      >
+        <div {...other} style={chipStyle}>
```

Comment:
I think that we should use `{...other}` on the root component: `EnhancedButton `.